### ISSUE TITLE
Removing purge key from app settings and uploading to server does not disable purging

### ISSUE THREAD

**Issue by sugat009** [medic/cht-conf#643]
**Describe the bug**
This issue is related to #642. This is mostly related to the upload part. After removing the `purge` key manually from `app_settings.json` and uploading it to the server. It doesn't get updated in the settings in the database.

**To Reproduce**
1. Create a purge.js file and add your purge logic
2. Run the command `cht compile-app-settings upload-app-settings`
3. Purging should be enabled
4. Manually remove the `purge` key from the `app_settings.json` file
5. Run the command `cht upload-app-settings`
6. Check if purging is disabled

**Expected behavior**
Purging should be disabled if the `purge` key is removed from the `app_settings.json` file and uploaded to the server.

**Logs**
If applicable, include the cht-conf output, server logs, and/or browser logs.

**Screenshots**

**Environment**
- Instance: Local
- Client platform: Linux
- cht-conf version: 4.1.1
- cht-core version: 4.13

**Additional context**
A workaround as described in #642 is to make the `purging.js` file blank and then, compile and upload the settings works but it creates a key in the settings in the database with an empty object which is probably not ideal.

**Comment by jkuester** [comment]
> While I understand the need for targeted updates, would the overwrite flag (which does a drop-in replacement for the passed request body) offer a more intuitive experience? 

IMHO this is `100%` correct.  It feels _very_ un-intuitive that adding a `purge` section enables purging, but removing the same will not disable purging.  _However_, my guess is this functionality was written this way to try to mitigate the issues with using both cht-conf and the App Management UI to configure an instance. (Basically, there are some settings that can be set in the App Management UI that affect the same `settings` doc that gets updated by the `cht-conf upload-settings` action.) In the case where the setting was configured via that App Managment UI and nothing was set in the cht-conf settings, then I guess the idea was not to blow away the existing configuration on the server from App Management when running the cht-conf upload-settings....) @dianabarsan does that sound correct? 

I think we all agree that allowing the same settings to be changed in App Managment and in cht-conf is a bad idea, but that I probably something we cannot change until CHT 5.0.  In the meantime, I am not sure what the best approach would be... 🤔

**Comment by dianabarsan** [comment]
I think Josh is right about the decision for cht-conf to not overwrite missing keys, it's exactly that changes could have been made through the UI and there was no expectation of rigid source control for configs way back when (in order to know the config of an instance, you'd need to login and check that instance). 

Is there a way that the key can be overwritten so that the server disables purging?

**Comment by sugat009** [comment]
yes, to disable purging, the `purge` key needs to be set to an empty object like `"purge": {}` in the settings.

**Comment by dianabarsan** [comment]
so it's either we document this requirement or .... we make an exception for rule compilation settings for purging specifically, which might make things really confusing.

**Comment by jkuester** [comment]
My vote is to document using `"purge": {}` for now and then delete the admin app in CHT 5.0 and revisit this. 😅  👍

**Comment by sugat009** [comment]
+1 for documenting this behavior and revisiting it later.

**Comment by sugat009** [comment]
I've created a ticket in `cht-docs`. ⬆️

**Comment by Akshanshkaushal** [comment]
@sugat009 I would like to work upon this issue. Could you please assign this?
